src/forces.f90

```diff
@@ -256,267 +268,626 @@ subroutine force(ux1,uy1,ep1)
+         if (ierror /= 0) then
+            call decomp_2d_abort(ierror, &
+                 "Error in MPI_ALLREDUCE!")
+         end if
+         call MPI_ALLREDUCE(MPI_IN_PLACE, convy, 1, real_type, MPI_SUM, MPI_COMM_WORLD, code)
+         if (ierror /= 0) then
+            call decomp_2d_abort(ierror, &
+                 "Error in MPI_ALLREDUCE!")
+         end if
+         call MPI_ALLREDUCE(MPI_IN_PLACE, pressx, 1, real_type, MPI_SUM, MPI_COMM_WORLD, code)
+         if (ierror /= 0) then
+            call decomp_2d_abort(ierror, &
+                 "Error in MPI_ALLREDUCE!")
+         end if
+         call MPI_ALLREDUCE(MPI_IN_PLACE, pressy, 1, real_type, MPI_SUM, MPI_COMM_WORLD, code)
+         if (ierror /= 0) then
+            call decomp_2d_abort(ierror, &
+                 "Error in MPI_ALLREDUCE!")
+         end if
+         call MPI_ALLREDUCE(MPI_IN_PLACE, stressx, 1, real_type, MPI_SUM, MPI_COMM_WORLD, code)
+         if (ierror /= 0) then
+            call decomp_2d_abort(ierror, &
+                 "Error in MPI_ALLREDUCE!")
+         end if
+         call MPI_ALLREDUCE(MPI_IN_PLACE, stressy, 1, real_type, MPI_SUM, MPI_COMM_WORLD, code)
+         if (ierror /= 0) then
+            call decomp_2d_abort(ierror, &
+                 "Error in MPI_ALLREDUCE!")
+         end if
+   
+         pressx = pressx/dt
+         pressy = pressy/dt 
+         
+         xDrag_mean = xmom + convx - pressx -stressx
+         yLift_mean = ymom + convy - pressy -stressy
+         
+         if (nrank .eq. 0) then
+             write(38,*) t,xDrag_mean,yLift_mean 
+             call flush(38)
+         endif
+         if (mod(itime, icheckpoint).eq.0) then
+             if (nrank .eq. 0) then
+               write(filename,"('forces.dat',I7.7)") itime
+               call execute_command_line("cp forces.dat " //filename)
+             endif
+         endif
        enddo
-
-       !Edited by F. Schuch
-       xDrag_mean = sum(xDrag(:))/real(nz,mytype)
-       yLift_mean = sum(yLift(:))/real(nz,mytype)
-
-       !     if ((itime==ifirst).or.(itime==0)) then
-       !        if (nrank .eq. 0) then
-       !        write(filename,"('aerof',I1.1)") iv
-       !        open(38+(iv-1),file=filename,status='unknown',form='formatted')
-       !        endif
-       !     endif
-       if (nrank .eq. 0) then
-          write(38,*) t,xDrag_mean,yLift_mean
-          flush(38)
-       endif
-       if (mod(itime, icheckpoint).eq.0) then
-          if (nrank .eq. 0) then
-             write(filename,"('forces.dat',I7.7)") itime
-             call execute_command_line ("cp forces.dat "//filename)
-          endif
-       endif
-    enddo
-
+    else
+       write(*,*) '==========================================================='
+       write(*,*) 'Error: Invalid value for the parameter i2dsim'
+       write(*,*) '==========================================================='
+       call MPI_ABORT(MPI_COMM_WORLD,code,ierror)
```

Comment:
Use `decomp_2d_abort` to abort the simulation.

Currently, in the PR, all ranks will print to stdout before calling `MPI_ABORT`, this is not a very good idea.